### PR TITLE
Add OpenSearch cluster infrastructure

### PR DESCRIPTION
**Pull request by acm19**
Adds OpenSearch cluster that can be deployed for development purposes.

Resolves: https://github.com/acm19/aws-request-signing-apache-interceptor/issues/73

### Pull Request Checklist:

- [x] I have added a contribution line at the top of [CHANGELOG.md](CHANGELOG.md).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -11,7 +11,7 @@ charset = utf-8
 trim_trailing_whitespace = true
 insert_final_newline = true
 
-[*.xml]
+[*.{xml,json}]
```
Comment (acm19) [inline .editorconfig:14]: Add `yml` and `yaml` as well.

File: infra/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+aws-cdk-lib==2.34.2
```
Comment (acm19) [inline infra/requirements.txt:1]: Update to latest before merge.

File: infra/Makefile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+.PHONY: synth
+.SILENT: synth
+synth: node_modules
+	./node_modules/.bin/cdk synth
+
+.PHONY: bootstrap
+.SILENT: bootstrap
+bootstrap:
+	./node_modules/.bin/cdk deploy
```
Comment (acm19) [inline infra/Makefile:26]: Change by `bootstrap` command.

File: infra/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+make .venv
+```
+
+After the init process completes and the virtualenv is created, you can use the following step to activate your virtualenv.
+
+```
+source .venv/bin/activate
+```
```
Comment (dblock) [inline infra/README.md:16]: Consider using [Pipenv](https://pipenv.pypa.io/en/latest/) as an alternative, IMO it's a lot simpler and behaves like other Rubygems/npm package managers.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -46,6 +46,10 @@ Implement your feature or bug fix.
 
 Make sure that `mvn test` completes without errors.
 
+## Use real infrastructure
+
+In case you need to test your feature against real infrastructure you can easily deploy a working cluster following [these instructions](infra/README.md).
```
Comment (dblock) [inline CONTRIBUTING.md:51]: Nit: To run integration tests against real infrastructure ...

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -46,6 +46,10 @@ Implement your feature or bug fix.
 
 Make sure that `mvn test` completes without errors.
 
+## Use real infrastructure
```
Comment (dblock) [inline CONTRIBUTING.md:49]: I think this is "integration tests" rather than about "real infrastructure"?

File: infra/README.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
+
+# OpenSearch cluster to test the interceptor at development time
```
Comment (dblock) [inline infra/README.md:2]: I would just say "Integration Tests" and "This project creates an OpenSearch cluster on AWS for development purposes ..."

**Review by dblock — APPROVED** [review]
Looks great! I have some nitpicks on the nomenclature (I think this is all about "creating an OpenSearch cluster on AWS to run integration tests"), feel free to edit/not/merge.

File: infra/README.md (hunk 4 above)
Comment (acm19) [reply infra/README.md:16]: As this very simple project I prefer to leave it like this. I don't need to install the extra package and the version is updated by `apt-get`. I also have the impression that Python version management world is quite divided and opinionated, I don't care much, I'm happy to make the change in the future if we find it to be more convenient but I'll carry on as is this time.

**Comment by acm19** [comment]
> Looks great! I have some nitpicks on the nomenclature (I think this is all about "creating an OpenSearch cluster on AWS to run integration tests"), feel free to edit/not/merge.

I appreciate these kind of comments.

**Pull request merged**